### PR TITLE
Initial Fix of PED Implementation Bugs

### LINKED ISSUE #123: [PE-D] set V to a value works but no circuit diagram displayed
Set R and Set C works as expected, an acknowledgement message was shown and a circuit diagram displayed

Set V only shows the acknowledge message

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: teachyourselfcoding/ped#1

### LINKED ISSUE #136: [PE-D] Suggestions for adding new components
A suggestion for adding new components in the circuit:
may change the value in "Nice, added a 400.0 Ω" to the value of added components, but not the sum. (Or maybe there is a small bug?)

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: ZhongNingmou/ped#2

### LINKED ISSUE #139: [PE-D] Overall suggestions
1. Maybe you can make CLIrcuit Assistant automatically print out the help diagram at the beginning.

2. Your summary diagram seperate commands and parameters, but I think that is a little bit confused for a new user. Maybe you can try to make some editions.

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: ZhongNingmou/ped#5

### LINKED ISSUE #140: [PE-D] LOGO position incorrect
Seems the LOGO is not being printed as excpected.
I am not sure whether the team intended to make it so.

-------------
Labels: `severity.VeryLow` `type.FeatureFlaw`
original: Lee-Juntong/ped#1

### LINKED ISSUE #145: [PE-D] Adding a resistor is actually change the resistor to the equivalent resistor after the adding operation
Although technically the circuit is equivalent to adding the parallel resistor there, the printed information is not quite correct since I did not add a 3.333333Ω resistor.

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: Lee-Juntong/ped#6

### LINKED ISSUE #146: [PE-D] No error message for invalid set in logic gate
When I set the value of B to something invalid, no error message provided.

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: Lee-Juntong/ped#7

### LINKED ISSUE #148: [PE-D] Unhandle exception
This happens when I try to add a parallel resistor to a 0 Ω resistor

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: Lee-Juntong/ped#9

### LINKED ISSUE #149: [PE-D] Unhandled exception
similar bug as the previous one

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: Lee-Juntong/ped#10

### LINKED ISSUE #150: [PE-D] Unhandled exception
Similar to the previous bugs

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: Lee-Juntong/ped#11

### LINKED ISSUE #151: [PE-D] Wrong output for xnor
The correct output should be 1 in this case

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: Lee-Juntong/ped#12

### LINKED ISSUE #152: [PE-D] xnor gate not working properly
-------------
Labels: `severity.High` `type.FunctionalityBug`
original: Lee-Juntong/ped#13

### LINKED ISSUE #153: [PE-D] nor gate not working
-------------
Labels: `severity.High` `type.FunctionalityBug`
original: Lee-Juntong/ped#14

### PR DESCRIPTION
**Pull request by hughjazzman**
From #155:

* Closes #123 
* Closes #136, closes #145
* Part of #139 
* Closes #140
* Part of #146
* Fixes #148, fixes #149, fixes #150
* Fixes #151, fixes #152, fixes #153

### CODE REVIEW

File: src/main/java/seedu/duke/model/template/BooleanTemplate.java [hunk 1]
```diff
@@ -138,9 +144,11 @@ private String getInputEquation(int index) throws DukeException {
 
         // Odd index - first input, even index, 2nd input
         if (index % 2 == 1) {
-            equation += gate.isSetInput() ? gate.getInput() : "?";
+            String input = gate.getInput() ? "1" : "0";
+            equation += gate.isSetInput() ? input : "?";
         } else {
-            equation += gate.isSetSecondInput() ? gate.getSecondInput() : "?";
+            String input = gate.getInput() ? "1" : "0";
+            equation += gate.isSetSecondInput() ? input : "?";
```
Comment (sevenseasofbri) [inline src/main/java/seedu/duke/model/template/BooleanTemplate.java]: ` String input = gate.getInput() ? "1" : "0";` can be extracted out and placed before the if-else block?

File: src/test/java/seedu/duke/model/gates/GateTest.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
+package seedu.duke.model.gates;
```
Comment (sevenseasofbri) [inline src/test/java/seedu/duke/model/gates/GateTest.java:1]: OOf this was not so trivial after all hahaha

**Review by sevenseasofbri — APPROVED** [review]
LGTM!!

File: src/main/java/seedu/duke/model/template/BooleanTemplate.java [hunk 3]
```diff
@@ -138,9 +144,11 @@ private String getInputEquation(int index) throws DukeException {
 
         // Odd index - first input, even index, 2nd input
         if (index % 2 == 1) {
-            equation += gate.isSetInput() ? gate.getInput() : "?";
+            String input = gate.getInput() ? "1" : "0";
+            equation += gate.isSetInput() ? input : "?";
         } else {
-            equation += gate.isSetSecondInput() ? gate.getSecondInput() : "?";
+            String input = gate.getInput() ? "1" : "0";
```
Comment (hughjazzman) [inline src/main/java/seedu/duke/model/template/BooleanTemplate.java]:
```suggestion
            String input = gate.getSecondInput() ? "1" : "0";
```

File: src/main/java/seedu/duke/model/template/BooleanTemplate.java (hunk 1 above)
Comment (hughjazzman) [reply src/main/java/seedu/duke/model/template/BooleanTemplate.java]: 😮You spotted a bug!

Comment (sevenseasofbri) [reply src/main/java/seedu/duke/model/template/BooleanTemplate.java]: omg hahahaha unbeknownst to me

**Pull request merged**